libs/designer-v2/src/lib/core/parsers/BJSWorkflow/BJSDeserializer.ts

```diff
@@ -359,6 +359,14 @@ const isIfAction = (action: LogicAppsV2.ActionDefinition): action is LogicAppsV2
   return equals(action?.type, 'if');
 };
 
+const isAgentCondition = (action: LogicAppsV2.AgentCondition | LogicAppsV2.McpClient): action is LogicAppsV2.AgentCondition => {
+  return !action?.type || !equals(action?.type, 'mcpclienttool');
+}
+
+const isMcpClient = (action: LogicAppsV2.AgentCondition | LogicAppsV2.McpClient): action is LogicAppsV2.McpClient => {
```

Comment:
Left this comment in the other PR too. Do we need two checks here? Can we not just use `if(isMcpClient()) else {}`